src/workspaces/workspaces.service.ts

```diff
@@ -70,4 +86,153 @@ export class WorkspacesService {
     }
     return userWorkspace.workspace
   }
+
+  async update(userId:string, workspaceId:string, updateWorkspaceDto: UpdateWorkspaceDto){
+
+
+    const workspace = await this.workspaceRepository.findOne({
+        where: {
+          id: workspaceId
+        }
+    })
+
+    if(!workspace){
+      throw new NotFoundException("Workspace not found")
+    }
+
+    const userWorkspace = await this.userWorkspaceRepository.findOne({
+      where: {
+        userId,
+        workspaceId
+      }
+    })
+
+    if(!userWorkspace) {
+      throw new ForbiddenException('Dear User you are not allowed to edit this workspace')
+    }
+
+
+    await this.workspaceRepository.update({id: workspaceId},{name: updateWorkspaceDto.name})
+    return await this.workspaceRepository.findOne({
+      where: {
+        id:workspaceId
+    }})
+  }
+
+  async inviteUser(userId: string,workspaceId: string, payload:InviteUserDto){
+     const workspace = await this.workspaceRepository.findOne({
+        where: {
+          id: workspaceId
+        }
+    })
+
+    if(!workspace){
+      throw new NotFoundException("Workspace not found")
+    }
+
+    // Check if you are the owner of the workspace 
+
+      const userWorkspace = await this.userWorkspaceRepository.findOne({
+      where: {
+        userId,
+        workspaceId,
+        role: UserRole.ADMIN
+      }
+    })
+
+    if(!userWorkspace) {
+      throw new ForbiddenException('Dear User you are not allowed to Invite User to this workspace')
+    }
```

Comment:
move these verification inside guards. You can add a guard like `@Permission(Role.Owner)`